wire-ios-request-strategy/Sources/Payloads/Processing/ConversationEventPayloadProcessor.swift

```diff
@@ -0,0 +1,777 @@
+            return nil
+        }
+
+        var created = false
+        let conversation = ZMConversation.fetchOrCreate(
+            with: conversationID,
+            domain: payload.qualifiedID?.domain,
+            in: context,
+            created: &created
+        )
+
+        conversation.conversationType = .`self`
+        conversation.domain = BackendInfo.isFederationEnabled ? payload.qualifiedID?.domain : nil
+        conversation.needsToBeUpdatedFromBackend = false
+        conversation.isPendingMetadataRefresh = false
+
+        updateMetadata(from: payload, for: conversation, context: context)
+        updateMembers(from: payload, for: conversation, context: context)
+        updateConversationTimestamps(for: conversation, serverTimestamp: serverTimestamp)
+
+        return conversation
+    }
+
+    @discardableResult
+    func updateOrCreateOneToOneConversation(
+        from payload: Payload.Conversation,
+        in context: NSManagedObjectContext,
+        serverTimestamp: Date,
+        source: Source
+    ) -> ZMConversation? {
+        guard
+            let conversationID = payload.id ?? payload.qualifiedID?.uuid,
+            let rawConversationType = payload.type
+        else {
+            Logging.eventProcessing.error("Missing conversation or type in 1:1 conversation payload, aborting...")
+            return nil
+        }
+
+        let conversationType = BackendConversationType.clientConversationType(rawValue: rawConversationType)
+
+        guard
+            let otherMember = payload.members?.others.first,
+            let otherUserID = otherMember.id ?? otherMember.qualifiedID?.uuid
+        else {
+            let conversation = ZMConversation.fetch(with: conversationID, domain: payload.qualifiedID?.domain, in: context)
+            // TODO: use conversation type from the backend once it returns the correct value
+            conversation?.conversationType = self.conversationType(for: conversation, from: conversationType)
+            conversation?.needsToBeUpdatedFromBackend = false
+            return conversation
+        }
+
+        let otherUser = ZMUser.fetchOrCreate(with: otherUserID, domain: otherMember.qualifiedID?.domain, in: context)
+
+        var conversation: ZMConversation
+        if let existingConversation = otherUser.connection?.conversation {
+            existingConversation.mergeWithExistingConversation(withRemoteID: conversationID)
+            conversation = existingConversation
+        } else {
+            conversation = ZMConversation.fetchOrCreate(with: conversationID, domain: payload.qualifiedID?.domain, in: context)
+            otherUser.connection?.conversation = conversation
+        }
+
+        conversation.remoteIdentifier = conversationID
+        conversation.domain = BackendInfo.isFederationEnabled ? payload.qualifiedID?.domain : nil
+
+        // TODO: use conversation type from the backend once it returns the correct value
```

Comment:
is it still relevant ? if yes, could we link it to a jira ticket or solve it already

Reply:
Yeah it's still a problem. I'll see if backend has a ticket for this.

Reply:
Actually this is now the way to do it, according to https://wearezeta.atlassian.net/browse/WPB-3560. So I'll remove the TODO.